sticht/rollbacks/base.py

```diff
@@ -0,0 +1,235 @@
+                    f'{len(failing)} of {len(self.slo_watchers)} SLOs are failing:\n',
+                ]
+                for slo_watcher in failing:
+                    slo_text_components.append(f'{slo_watcher.label}\n')
+            else:
+
+                unknown = [
+                    w
+                    for w in self.slo_watchers
+                    if w.bad_before_mark is None or w.bad_after_mark is None
+                ]
+                bad_before_mark = [w for w in self.slo_watchers if w.bad_before_mark]
+                slo_text_components = []
+                if len(unknown) > 0:
+                    slo_text_components.extend(
+                        [
+                            Emoji(':thinking_face:'),
+                            f'{len(unknown)} SLOs are missing data:\n',
+                        ],
+                    )
+                    for slo_watcher in unknown:
+                        slo_text_components.append(f'{slo_watcher.label}\n')
+
+                if len(bad_before_mark) > 0:
+                    slo_text_components.extend(
+                        [
+                            Emoji(':grimacing:'),
+                            f'{len(bad_before_mark)} SLOs were failing before deploy, and will be ignored:\n',
+                        ],
+                    )
+                    for slo_watcher in bad_before_mark:
+                        slo_text_components.append(f'{slo_watcher.label}\n')
+
+                remaining = len(self.slo_watchers) - len(unknown) - len(bad_before_mark)
+
+                if remaining == len(self.slo_watchers):
+                    slo_text_components = [
+                        Emoji(':ok_hand:'),
+                        f'All {len(self.slo_watchers)} SLOs are currently passing.',
+                    ]
+                else:
+                    if remaining > 0:
+                        slo_text_components.append(
+                            f'The remaining {remaining} SLOs are currently passing.',
+                        )
+
+            if summary:
+                # For summary, only display emojis.
+                if self.is_terminal_state(self.state):
+                    return ''
+                else:
+                    return ' '.join(
+                        [c for c in slo_text_components if isinstance(c, Emoji)],
+                    )
+            else:
+                # Display all text for non-summary mode, but hide Emojis if we're in a terminal state, to prevent
+                # things like :alert: from blinking until the end of time.
+                if self.is_terminal_state(self.state):
+                    return ' '.join(
+                        [c for c in slo_text_components if not isinstance(c, Emoji)],
+                    )
+                else:
+                    return ' '.join(slo_text_components)
+        else:
+            return ''
+
+    def get_metric_text(self, summary: bool) -> str:
```

Comment:
this is what'll power the query-related messages i alluded to above (where this function is called)